### PR TITLE
Fix text not being occluded by the globe when overlapmode=always

### PR DESCRIPTION
**Pull request by kubapelc**
This PR fixes [#4802](https://github.com/maplibre/maplibre-gl-js/issues/4802) and adds render tests for this behaviour. The text and icon tests would fail before this PR, but lines seem to already have been properly occluded by the globe regardless of overlap mode.

### CODE REVIEW

File: src/symbol/placement.ts [hunk 1]
```diff
@@ -809,7 +809,10 @@ export class Placement {
             if (symbolInstance.crossTileID === 0) throw new Error('symbolInstance.crossTileID can\'t be 0');
             if (bucket.bucketInstanceId === 0) throw new Error('bucket.bucketInstanceId can\'t be 0');
 
-            this.placements[symbolInstance.crossTileID] = new JointPlacement(placeText || alwaysShowText, placeIcon || alwaysShowIcon, offscreen || bucket.justReloaded);
+            const textOccluded = placedGlyphBoxes && placedGlyphBoxes.occluded;
```
Comment (HarelM) [inline src/symbol/placement.ts:812]: use "?." maybe?

File: src/symbol/placement.ts [hunk 2]
```diff
@@ -809,7 +809,10 @@ export class Placement {
...
+            // Do not show text or icons that are occluded by the globe, even if overlap mode is 'always'!
+            const iconOccluded = placedIconBoxes && placedIconBoxes.occluded;
+            this.placements[symbolInstance.crossTileID] = new JointPlacement((placeText || alwaysShowText) && !textOccluded, (placeIcon || alwaysShowIcon) && !iconOccluded, offscreen || bucket.justReloaded);
```
Comment (HarelM) [inline src/symbol/placement.ts:815]: I would extract the logic here to variables to make is more readable.

**Comment by HarelM** [comment]
I've added a few nit-picking, otherwise, this looks great, thanks!

File: src/symbol/placement.ts (hunk 1 above)
Comment (kubapelc) [reply src/symbol/placement.ts]: Done!

File: src/symbol/placement.ts (hunk 2 above)
Comment (kubapelc) [reply src/symbol/placement.ts]: Done!

**Review by HarelM — APPROVED** [review]

**Pull request merged**